### PR TITLE
Properly Open Bookmark Modal from Study Mode

### PR DESCRIPTION
**Pull request by al-imam**
## Summary

Integrates BookmarkModal into the VerseActionModalContainer system, enabling bookmarks to be opened from study mode with proper state management and back navigation support.

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [x] ♻️ Refactoring (no functional changes)

## Scope Confirmation

- [x] This PR addresses **one** feature/fix only

## Test Plan

- [x] Manual testing performed

**Testing steps:**

1. Open a verse in study mode
2. Click the bookmark action button
3. Verify BookmarkModal opens with correct verse information
4. Verify back button appears and returns to study mode with proper state restored
5. Test bookmarking from regular reader mode (not study mode)
6. Verify existing bookmark functionality still works correctly

### Edge Cases Verified

- [x] Study mode state properly restored when closing bookmark modal
- [x] Back button only shows when opened from study mode
- [x] Bookmark modal works for both translation and Arabic views
- [x] Redux state properly managed for modal lifecycle

| Before | After |
|--------|-------|
| image | image |

## Reviewer Notes

The main changes:
- **BookmarkAction.tsx**: Refactored from local modal state to Redux-based modal management (removed `useState`, dynamic import)
- **BookmarkModal.tsx**: New component wrapping SaveBookmarkModal for VerseActionModalContainer
- **VerseActionModalContainer**: Added handling for `SAVE_BOOKMARK` modal type
- **SaveBookmarkModal**: Added `onBack` and `wasOpenedFromStudyMode` props for study mode integration
- **Redux**: Added new `SAVE_BOOKMARK` modal type and `openBookmarkModal` action

### CODE REVIEW

File: src/components/QuranReader/VerseActionModalContainer/index.tsx [hunk 1]
```diff
@@ -173,6 +174,17 @@ const VerseActionModalContainer: React.FC = () => {
...
+  if (modalType === VerseActionModalType.SAVE_BOOKMARK && verse) {
+    return (
+      <BookmarkModal
+        verse={verse}
+        wasOpenedFromStudyMode={wasOpenedFromStudyMode}
+        onClose={handleClose}
+        onBack={handleBackToStudyMode}
+      />
+    );
+  }
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/QuranReader/VerseActionModalContainer/index.tsx:186]: `useBatchedCountRangeNotes` hook is created in a way so we can call it how many time we want.

File: src/components/Verse/BookmarkAction.tsx [hunk 2]
```diff
@@ -56,15 +74,40 @@ const BookmarkAction: React.FC<Props> = ({ verse, isTranslationView }): JSX.Elem
         }_verse_actions_menu_bookmark_open`,
       );
 
-      setIsModalOpen(true);
+      // Determine if opened from study mode
+      const openedFromStudyMode = isInsideStudyMode || (isStudyModeOpen && !isSsrMode);
+
+      dispatch(
+        openBookmarkModal({
+          verseKey: `${verse.chapterId}:${verse.verseNumber}`,
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/Verse/BookmarkAction.tsx]: done

File: src/components/Verse/BookmarkAction.tsx [hunk 3]
```diff
@@ -1,39 +1,50 @@
...
+import { openBookmarkModal } from '@/redux/slices/QuranReader/verseActionModal';
 import Verse from '@/types/Verse';
 import { logButtonClick } from '@/utils/eventLogger';
 
-const SaveBookmarkModal = dynamic(() => import('./SaveBookmarkModal'), { ssr: false });
-
 interface Props {
   verse: Verse;
   isTranslationView: boolean;
   onActionTriggered?: () => void;
```
[the remark this answers is not in the document]
Comment (al-imam) [reply src/components/Verse/BookmarkAction.tsx:29]: done
